bin/runtime-common/src/refund_relayer_extension.rs

```diff
@@ -265,46 +342,51 @@ where
 		_info: &DispatchInfoOf<Self::Call>,
 		_len: usize,
 	) -> TransactionValidity {
-		if let Some(calls) = self.expand_call(call) {
-			for nested_call in calls {
-				nested_call.check_obsolete_submit_finality_proof()?;
-				nested_call.check_obsolete_submit_parachain_heads()?;
-				nested_call.check_obsolete_call()?;
+		// this is the only relevant line of code for the `pre_dispatch`
+		//
+		// we're not calling `validato` from `pre_dispatch` directly because of performance
+		// reasons, so if you're adding some code that may fail here, please check if it needs
+		// to be added to the `pre_dispatch` as well
+		let parsed_call = self.parse_and_check_for_obsolete_call(call)?;
+
+		// the following code just plays with transaction priority and never returns an error
+		let mut valid_transaction = ValidTransactionBuilder::default();
+		if let Some(parsed_call) = parsed_call {
+			// we give delivery transactions some boost, that depends on number of messages inside
+			let messages_call_info = parsed_call.messages_call_info();
+			if let MessagesCallInfo::ReceiveMessagesProof(ref delivery_info) = messages_call_info {
+				// compute total number of messages in transaction
+				let bundled_range = &delivery_info.0.bundled_range;
+				let bundled_messages = bundled_range
+					.end()
+					.checked_sub(*bundled_range.start())
+					.map(|d| d.saturating_add(1))
+					.unwrap_or(0);
+
+				// a quick check to avoid invalid high-priority transactions
+				if bundled_messages <= Runtime::MaxUnconfirmedMessagesAtInboundLane::get() {
+					let priority_boost = crate::priority_calculator::compute_priority_boost::<
+						Priority,
+					>(bundled_messages);
+					valid_transaction = valid_transaction.priority(priority_boost);
+				}
```

Comment:
I'd leave it here, because this code belongs to the signed extension that deals with the refunds and priorities (since this PR), not the info about the call. I could extract to the separate function here if you prefer

Reply:
Ok. Makes sense. But could we move at least this piece:

```
				let bundled_range = &delivery_info.0.bundled_range;
				let bundled_messages = bundled_range
					.end()
					.checked_sub(*bundled_range.start())
					.map(|d| d.saturating_add(1))
					.unwrap_or(0);
```

to a method defined on `BaseMessagesProofInfo` ? Something like `bundled_range_len()`.